mlflow/store/dynamodb_store.py

```diff
@@ -0,0 +1,707 @@
+
+
+def _entity_to_dict(obj):
+    return {k: None if v == '' else v for k, v in dict(obj).items()}
+
+
+def _filter_view_type(d, view_type=None):
+    return not view_type or view_type == ViewType.ALL or \
+        (view_type == ViewType.ACTIVE_ONLY and
+            d.get('lifecycle_stage') == RunInfo.ACTIVE_LIFECYCLE) or \
+        (view_type == ViewType.DELETED_ONLY and
+            d.get('lifecycle_stage') == RunInfo.DELETED_LIFECYCLE)
+
+
+def _filter_experiment(experiments, view_type=None, name=None):
+    return [e for e in experiments if _filter_view_type(e, view_type)
+            and (not name or e['name'] == name)]
+
+
+def _filter_run(runs, view_type=None, experiment_id=None):
+    return [r for r in runs if _filter_view_type(r, view_type)
+            and (not experiment_id or r['experiment_id'] == experiment_id)]
+
+
+class DynamodbStore(AbstractStore):
+    EXPERIMENT_TABLE = "experiment"
+    RUN_TABLE = "run"
+    METRICS_TABLE = "run_metric"
+    PARAMS_TABLE = "run_param"
+    TAGS_TABLE = "run_tag"
+
+    def __init__(self, dynamodb_resource=None, table_prefix=None,
+                 use_gsi=True, use_projections=True):
+        """
+        Create a new DynamodbStore with artifact root URI.
+        """
+        super(DynamodbStore, self).__init__()
+        self.dynamodb_resource = dynamodb_resource or _default_dynamodb_resource()
+        self.table_prefix = table_prefix or _default_table_prefix()
+        self.use_gsi = use_gsi
+        self.use_projections = use_projections
+
+    def _get_dynamodb_resource(self):
+        return self.dynamodb_resource
+
+    def _list_experiments(self, view_type=None, name=None):
+        dynamodb = self._get_dynamodb_resource()
+        table_name = '_'.join([self.table_prefix, DynamodbStore.EXPERIMENT_TABLE])
+        table = dynamodb.Table(table_name)
+
+        # Filter on active/deleted with optional name
+        condition = None
+        if self.use_gsi and view_type and view_type != ViewType.ALL:
+            if view_type == ViewType.ACTIVE_ONLY:
+                condition = Key('lifecycle_stage').eq(RunInfo.ACTIVE_LIFECYCLE)
+            elif view_type == ViewType.DELETED_ONLY:
+                condition = Key('lifecycle_stage').eq(RunInfo.DELETED_LIFECYCLE)
+            if name:
+                condition = And(condition, Key('name').eq(name))
+            response = table.query(
+                IndexName='LifeCycleStage',
+                KeyConditionExpression=condition,
+                ReturnConsumedCapacity='TOTAL',
+            )
+        elif name:
+            condition = Key('name').eq(name)
+            response = table.scan(
+                FilterExpression=condition,
+                ReturnConsumedCapacity='TOTAL',
+            )
+        else:
+            response = table.scan(
+                ReturnConsumedCapacity='TOTAL',
+            )
+
+        items = []
+        if response['ResponseMetadata']['HTTPStatusCode'] == 200 and 'Items' in response:
```

Comment:
You should probably throw exception if status code != 200?

Reply:
Update all the places to raise an connection error when status code != 200.